### PR TITLE
chainhash: Remove dup code from hash funcs.

### PR DESCRIPTION
**Pull request by orthomind**
Removed some duplicate code by having HashH use HashB rather than duplicating its functionality.

### CODE REVIEW

**Comment by hypernoob** [comment]
HashFunc also has duplicate codes.

**Comment by orthomind** [comment]
Good point :)  I'll fix that up too, and make another change.  Don't merge yet!

**Comment by hypernoob** [comment]
I’d prefer naming the '[]byte' parameter 'b'

https://github.com/golang/go/wiki/CodeReviewComments#variable-names

**Comment by davecgh** [comment]
As far as `b` vs `data`, I'm fine with either, but the param names should match across the funcs and this PR, as it is with `data`, makes them match.

File: chaincfg/chainhash/hashfuncs_test.go [hunk 1]
```diff
@@ -70,3 +72,15 @@ func TestHashFuncs(t *testing.T) {
...
 	}
 }
+
+// TestHashFunc ensure HashFunc returns the correct hashed bytes for given
+// bytes
+func TestHashFunc(t *testing.T) {
+	for _, test := range hashTests {
+		h := fmt.Sprintf("%x", HashFunc([]byte(test.in)))
+		if h != test.out {
+			t.Errorf("HashB(%q) = %s, want %s", test.in, h, test.out)
```
Comment (davecgh) [inline chaincfg/chainhash/hashfuncs_test.go]: s/HashB/HashFunc/

**Pull request merged**